boa_parser/src/lexer/mod.rs

```diff
@@ -236,7 +240,12 @@ impl<R> Lexer<R> {
                 )),
                 '.' => {
                     if self.cursor.peek()?.map(|c| (b'0'..=b'9').contains(&c)) == Some(true) {
-                        NumberLiteral::new(next_ch as u8).lex(&mut self.cursor, start, interner)
+                        NumberLiteral::new(
+                            next_ch
+                                .try_into()
+                                .expect("an ascii char must not fail to convert"),
+                        )
+                        .lex(&mut self.cursor, start, interner)
```

Comment:
Here we could completeley avoid the conversion by doing `NumberLiteral::new(b'.').lex(&mut self.cursor, start, interner)`